server/src/main/java/org/opensearch/common/util/concurrent/ThreadContext.java

```diff
@@ -534,7 +534,17 @@ boolean isDefaultContext() {
      * by the system itself rather than by a user action.
      */
     public void markAsSystemContext() {
-        threadLocal.set(threadLocal.get().setSystemContext());
+        ThreadContextStruct threadContextStruct = threadLocal.get();
```

Comment:
I believe we should encapsulate the logic inside `ThreadContextStruct` (since this is the entry point for system context creation)

```suggestion
        threadLocal.set(threadLocal.get().setSystemContext(propagators));
```